tests/gui/test_orb.py

```diff
@@ -0,0 +1,208 @@
+
+
+def test_rom_interface_write_state_move_timer_to_exit_invalid():
+    rom = create_move_timer_to_exit_rom(bytearray([0x8c, 0xf4, 0x7c]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().move_timer_to_exit
+    rom_interface.write_state(State(False, None, False))
+    assert False is rom_interface.read_state().move_timer_to_exit
+
+
+def create_stop_timer_rom(jump_table: bytearray, stop_timer: bytearray) -> Rom:
+    rom = Rom(bytearray([0] * 0x50000))
+
+    rom.write(0x6014 - 4, bytearray([0x46, 0xBF, 0xFF, 0xA9]))
+    rom.write(0x6014, jump_table)
+    rom.write(0x6014 + len(jump_table), bytearray([0x88, 0xd0, 0x0b, 0x8c]))
+
+    rom.write(0x7FCF - 4, bytearray([0x07, 0x4C, 0xE7, 0xD5]))
+    rom.write(0x7FCF, stop_timer)
+    return rom
+
+
+def test_rom_interface_read_state_stop_timer_false():
+    rom = create_stop_timer_rom(bytearray([0x5B, 0xA9]), bytearray([0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().touch_game_timer_stops
+
+
+def test_rom_interface_read_state_stop_timer_true():
+    rom = create_stop_timer_rom(bytearray([0xBF, 0xBF]), bytearray([0xA9, 0x01, 0x8D, 0xF3, 0x05, 0x60]))
+    rom_interface = RomInterface(rom)
+    assert True is rom_interface.read_state().touch_game_timer_stops
+
+
+def test_rom_interface_read_state_stop_timer_invalid():
+    rom = create_stop_timer_rom(bytearray([0x5B, 0xA9]), bytearray([0xA9, 0x01, 0x8D, 0xF3, 0x05, 0x60]))
+    rom_interface = RomInterface(rom)
+    assert None is rom_interface.read_state().touch_game_timer_stops
+
+    rom = create_stop_timer_rom(bytearray([0xBF, 0xBF]), bytearray([0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF]))
+    rom_interface = RomInterface(rom)
+    assert None is rom_interface.read_state().touch_game_timer_stops
+
+
+def test_rom_interface_write_state_stop_timer_false():
+    rom = create_stop_timer_rom(bytearray([0xBF, 0xBF]), bytearray([0xA9, 0x01, 0x8D, 0xF3, 0x05, 0x60]))
+    rom_interface = RomInterface(rom)
+    assert True is rom_interface.read_state().touch_game_timer_stops
+    rom_interface.write_state(State(True, True, False))
+    assert False is rom_interface.read_state().touch_game_timer_stops
+
+
+def test_rom_interface_write_state_stop_timer_true():
+    rom = create_stop_timer_rom(bytearray([0x5B, 0xA9]), bytearray([0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().touch_game_timer_stops
+    rom_interface.write_state(State(False, False, True))
+    assert True is rom_interface.read_state().touch_game_timer_stops
+
+
+def test_rom_interface_write_state_stop_timer_invalid():
+    rom = create_stop_timer_rom(bytearray([0x5B, 0xA9]), bytearray([0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().touch_game_timer_stops
+    rom_interface.write_state(State(False, False, None))
```

Comment:
Either State should use the Optional keyword or None should not be considered as a valid state.